src/firebase_app/hooks/useFirestoreDataSource.ts

```diff
@@ -129,30 +129,33 @@ export function useFirestoreDataSource({
      * @category Firestore
      */
     function firestoreToCMSModel<M>(data: any, schema: ResolvedEntitySchema<M>, path: string): any {
-        return traverseValues(data,
-            schema.properties,
-            (value, property) => {
-                if (value === null)
-                    return null;
-
-                if (serverTimestamp().isEqual(value)) {
-                    return null;
-                }
-
-                if (value instanceof Timestamp && property.dataType === "timestamp") {
-                    return value.toDate();
-                }
-
-                if (value instanceof GeoPoint && property.dataType === "geopoint") {
-                    return new GeoPoint(value.latitude, value.longitude);
-                }
-
-                if (value instanceof DocumentReference && property.dataType === "reference") {
-                    return new EntityReference(value.id, getCMSPathFromFirestorePath(value.path));
+        const traverse = (input: any): any => {
+            if (input == null) return input;
+            if (serverTimestamp().isEqual(input)) {
+                return null;
+            }
+            if (input instanceof Timestamp) {
+                return input.toDate();
+            }
+            if (input instanceof GeoPoint) {
+                return new GeoPoint(input.latitude, input.longitude);
```

Comment:
I think those GeoPoint classes should be different, onse should be the one from Firestore and the other from FireCMS